packages/grafana-data/src/utils/scale.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Field,
+  Threshold,
+  ScaleCalculator,
+  ScaleMode,
+  GrafanaTheme,
+  GrafanaThemeType,
+  Scale,
+  ColorScheme,
+} from '../types';
+import { reduceField, ReducerID } from '../transformations';
+import { getColorFromHexRgbOrName } from './namedColorsPalette';
+import * as d3 from 'd3-scale-chromatic';
+import isNumber from 'lodash/isNumber';
+
+/**
+ * @param t Number in the range [0, 1].
+ */
+type colorInterpolator = (t: number) => string;
+
+export function getScaleCalculator(field: Field, theme?: GrafanaTheme): ScaleCalculator {
+  const themeType = theme ? theme.type : GrafanaThemeType.Dark;
+  const config = field.config || {};
+  const scale = config.scale;
+  if (!scale) {
+    return (value: number) => {
+      return {}; // NOOP
+    };
+  }
+  const absolute = scale.mode === ScaleMode.absolute;
+  if (absolute && !scale.scheme) {
+    return (value: number) => {
+      const threshold = getActiveThreshold(value, scale.thresholds);
+      return {
+        threshold,
+        color: getColorFromHexRgbOrName(threshold.color, themeType),
+      };
+    };
+  }
```

Comment:
should not absolute always use thresholds (no matter if a scheme is set) ?

Reply:
I waffled on this issue -- my thought for the logic here is that thresholds could set the meaning (threshold.state), but the scheme could do the color.

That said, the current [validateScheme](https://github.com/grafana/grafana/pull/20783/files#diff-d6f4d97b23eb30f6346d5e6e325c51adR97) code removes the scheme if the mode is not scheme, so the mode is not possible.